src/gui/clips/AutomationClipView.cpp

```diff
@@ -487,4 +496,51 @@ void AutomationClipView::scaleTimemapToFit( float oldMin, float oldMax )
 }
 
 
+
+
+bool AutomationClipView::splitClip(const TimePos pos)
+{
+	setMarkerEnabled(false);
+
+	const TimePos splitPos = m_initialClipPos + pos;
+
+	// Don't split if we slid off the Clip or if we're on the clip's start/end
+	// Cutting at exactly the start/end position would create a zero length
+	// clip (bad), and a clip the same length as the original one (pointless).
+	if (splitPos <= m_initialClipPos || splitPos >= m_initialClipEnd) { return false; }
+
+	m_clip->getTrack()->addJournalCheckPoint();
+	m_clip->getTrack()->saveJournallingState(false);
+
+	auto rightClip = new AutomationClip(*m_clip);
+	auto leftClip = new AutomationClip(*m_clip);
+
+	rightClip->clear();
+	leftClip->removeNodes(splitPos, m_initialClipEnd);
+
+	for (auto it = m_clip->getTimeMap().begin(); it != m_clip->getTimeMap().end(); ++it)
+	{
+		if (POS(it) >= pos)
+		{
+			rightClip->putValues(POS(it) - pos, INVAL(it), OUTVAL(it), false);
+		}
+	}
+	rightClip->putValue(0, m_clip->valueAt(pos));
+	leftClip->putValue(pos, m_clip->valueAt(pos));
+
+	leftClip->movePosition(m_initialClipPos);
+	leftClip->changeLength(splitPos - m_initialClipPos);
+
+	rightClip->movePosition(splitPos);
+	rightClip->changeLength(m_initialClipEnd - splitPos);
+
+	// For some reason, the new clips sometime randomly put themselves in record mode. This is a temportary fix which forces them to match the original clip.
+	rightClip->setRecording(m_clip->isRecording());
+	leftClip->setRecording(m_clip->isRecording());
```

Comment:
It looks like `AutomationClip`'s copy constructor doesn't initialize some of its member variables.
`m_dragging`, `m_isRecording`, and `m_lastRecordedValue` are all left uninitialized.

`m_dragging` should probably always be set to `false` rather than copied, but I'm not sure what would be correct for the other two.